src/kOS/AddOns/RemoteTech/RemoteTechAntennaModuleFields.cs

```diff
@@ -118,28 +117,31 @@ protected new Structure GetKSPFieldValue(StringValue suffixName)
 
         private Guid GetTargetGuid(Structure target)
         {
-            var str = target as StringValue;
-            if (str != null)
+            var api = RemoteTechHook.Instance;
+            if (target is StringValue)
             {
-                string targetString = str;
+                string targetString = target.ToString();
                 if (targetString.Equals(NoTargetString, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    return new Guid(NoTargetGuid);
+                    return api.GetNoTargetGuid();
                 }
                 else if (targetString.Equals(ActiveVesselString, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    return new Guid(ActiveVesselGuid);
-                }
-                else if (targetString.Equals(MissionControlString, StringComparison.InvariantCultureIgnoreCase))
-                {
-                    return new Guid(MissionControlGuid);
+                    return api.GetActiveVesselGuid();
                 }
                 else
                 {
-                    var body = FlightGlobals.Bodies.FirstOrDefault(b => b.bodyName.Equals(targetString, StringComparison.InvariantCultureIgnoreCase));
+                    IEnumerable<string> groundStations = api.GetGroundStations();
+                    foreach (var groundStation in groundStations) {
+                        if (targetString.Equals(groundStation)) {
```

Comment:
If switched to `Equals(groundStation, StringComparison.OrdinalIgnoreCase)` setting the ground station will no longer be case sensitive, which is preferred since most of kOS is case-insensitive.

I'll grab it in review, and I'll switch the other string comparisons in this section to us Oridinal which is faster than InvariantCulture